storm-core/src/jvm/org/apache/storm/container/cgroup/core/CpusetCore.java

```diff
@@ -0,0 +1,209 @@
+    }
+
+    public boolean isMemHardwall() throws IOException {
+        int output = Integer.parseInt(CgroupUtils.readFileByLine(CgroupUtils.getDir(this.dir, CPUSET_MEM_HARDWALL)).get(0));
+        return output > 0;
+    }
+
+    public int getMemPressure() throws IOException {
+        String output = CgroupUtils.readFileByLine(CgroupUtils.getDir(this.dir, CPUSET_MEMORY_PRESSURE)).get(0);
+        return Integer.parseInt(output);
+    }
+
+    public void setMemPressureEnabled(boolean flag) throws IOException {
+        CgroupUtils.writeFileByLine(CgroupUtils.getDir(this.dir, CPUSET_MEMORY_PRESSURE_ENABLED), String.valueOf(flag ? 1 : 0));
+    }
+
+    public boolean isMemPressureEnabled() throws IOException {
+        int output = Integer.parseInt(CgroupUtils.readFileByLine(CgroupUtils.getDir(this.dir, CPUSET_MEMORY_PRESSURE_ENABLED)).get(0));
+        return output > 0;
+    }
+
+    public void setMemSpreadPage(boolean flag) throws IOException {
+        CgroupUtils.writeFileByLine(CgroupUtils.getDir(this.dir, CPUSET_MEMORY_SPREAD_PAGE), String.valueOf(flag ? 1 : 0));
+    }
+
+    public boolean isMemSpreadPage() throws IOException {
+        int output = Integer.parseInt(CgroupUtils.readFileByLine(CgroupUtils.getDir(this.dir, CPUSET_MEMORY_SPREAD_PAGE)).get(0));
+        return output > 0;
+    }
+
+    public void setMemSpreadSlab(boolean flag) throws IOException {
+        CgroupUtils.writeFileByLine(CgroupUtils.getDir(this.dir, CPUSET_MEMORY_SPREAD_SLAB), String.valueOf(flag ? 1 : 0));
+    }
+
+    public boolean isMemSpreadSlab() throws IOException {
+        int output = Integer.parseInt(CgroupUtils.readFileByLine(CgroupUtils.getDir(this.dir, CPUSET_MEMORY_SPREAD_SLAB)).get(0));
+        return output > 0;
+    }
+
+    public void setSchedLoadBlance(boolean flag) throws IOException {
+        CgroupUtils.writeFileByLine(CgroupUtils.getDir(this.dir, CPUSET_SCHED_LOAD_BALANCE), String.valueOf(flag ? 1 : 0));
+    }
+
+    public boolean isSchedLoadBlance() throws IOException {
+        int output = Integer.parseInt(CgroupUtils.readFileByLine(CgroupUtils.getDir(this.dir, CPUSET_SCHED_LOAD_BALANCE)).get(0));
+        return output > 0;
+    }
+
+    public void setSchedRelaxDomainLevel(int value) throws IOException {
+        CgroupUtils.writeFileByLine(CgroupUtils.getDir(this.dir, CPUSET_SCHED_RELAX_DOMAIN_LEVEL), String.valueOf(value));
+    }
+
+    public int getSchedRelaxDomainLevel() throws IOException {
+        String output = CgroupUtils.readFileByLine(CgroupUtils.getDir(this.dir, CPUSET_SCHED_RELAX_DOMAIN_LEVEL)).get(0);
+        return Integer.parseInt(output);
+    }
+
+    public static int[] parseNums(String outputStr) {
+        char[] output = outputStr.toCharArray();
+        LinkedList<Integer> numList = new LinkedList<Integer>();
+        int value = 0;
+        int start = 0;
+        boolean isHyphen = false;
+        for (char ch : output) {
```

Comment:
can this be simplified to 
```
for (str in outputStr.split(",")) 
if (!str.contains("-"))
numList.add(Integer.valueOf(str))
else {
Integer start = Integer.valueOf(str.split("-")[0])
Integer end = Integer.valueOf(str.split("-")[1])
// Add from start to end to numList
}
```
It is based on my understanding of list format. Hope it covers the cases